src/remoteConnector.ts

```diff
@@ -766,4 +760,53 @@ export default class RemoteConnector extends Disposable {
 			this.logger.error('Failed to disable auto tunneling:', e);
 		}
 	}
+
+	public async checkRemoteConnectionSuccessful() {
+		const isRemoteExtensionHostRunning = async () => {
+			try {
+				// Invoke command from gitpot-remote extension to test if connection is successful
+				await vscode.commands.executeCommand('__gitpod.getGitpodRemoteLogsUri');
+				return true;
+			} catch {
+				return false;
+			}
+		};
+
+		const remoteUri = vscode.workspace.workspaceFile || vscode.workspace.workspaceFolders?.[0].uri;
+		if (vscode.env.remoteName === 'ssh-remote' && this.context.extension.extensionKind === vscode.ExtensionKind.UI && remoteUri) {
+			const [, sshDestStr] = remoteUri.authority.split('+');
+			const sshDest: { user: string; hostName: string } = JSON.parse(Buffer.from(sshDestStr, 'hex').toString('utf8'));
+			const gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
+			const host = new URL(gitpodHost).host;
+			const connectionSuccessful = sshDest.hostName.endsWith(host) && await isRemoteExtensionHostRunning();
+
+			const connectionInfo = this.context.globalState.get<SSHConnectionParams & { usingSSHGateway: boolean }>(sshDestStr);
+			if (connectionInfo) {
+				const kind = connectionInfo.usingSSHGateway ? 'gateway' : 'local-app';
+				if (connectionSuccessful) {
```

Comment:
do we consider successful a connection even if the user failed to enter the right password?

Reply:
Nope, it should send the `failed` status